hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java

```diff
@@ -460,7 +460,7 @@ public static OMRequest handleCreateBucketWithBucketLayoutDuringPreFinalize(
    * they do not understand.
    */
   @RequestFeatureValidator(
-      conditions = ValidationCondition.OLDER_CLIENT_REQUESTS,
+      conditions = ValidationCondition.OLDER_BUCKET_LAYOUT_CLIENT_REQUESTS,
```

Comment:
With Ozone 1.4.0, client supports bucket layout.  The feature validator is wrongly rejecting the request.

Reply:
Rejection does not happen in the annotation. The annotation determines when the method should be called. In this case this method is called when we get an older client request. That's why the method to check the condition is called [shouldApply](https://github.com/apache/ozone/blob/74d92c86be579e6d90535a13a276e1970ac644fc/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/validation/ValidationCondition.java#L49). The body of the method determines whether to reject or modify the request. The body of the method has a more specific check for specific client version.

Reply:
@errose28 it happens here https://github.com/apache/ozone/blob/74d92c86be579e6d90535a13a276e1970ac644fc/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/validation/ValidationCondition.java#L50

Reply:
We have multiple such validations, this method takes care but other methods don't.

Reply:
I agree. I also suggested a refactor, but in a follow up. The refactor done here is not good because it requires updates to two enums for each client layout version.